storage/innobase/dict/dict0dd.cc

```diff
@@ -1689,8 +1689,11 @@ void dd_copy_table_columns(const Alter_inplace_info *ha_alter_info,
   for (const auto old_col : old_table.columns()) {
     if (old_col->is_se_hidden() && !is_system_column(old_col->name().c_str()) &&
         (strcmp(old_col->name().c_str(), FTS_DOC_ID_COL_NAME) != 0)) {
-      /* Must be an alredy dropped column. */
-      ut_ad(dd_column_is_dropped(old_col));
+      /* TRUNCATE will erase INSTANT ADD/DROP information */
+      if (!first_row_version) {
+        /* Must be an alredy dropped column. */
+        ut_ad(dd_column_is_dropped(old_col));
+      }
```

Comment:
@altmannmarcelo 
the only time a col is set hidden (this has different meaning at server layer) in InnoDB is this

```
* Add hidden FTS_DOC_ID column */
    dd::Column *col = new_table.add_column();
    col->set_hidden(dd::Column::enum_hidden_type::HT_HIDDEN_SE);
    col->set_name(FTS_DOC_ID_COL_NAME);
    col->set_type(dd::enum_column_types::LONGLONG);
    col->set_nullable(false);
    col->set_unsigned(true);
    col->set_collation_id(1);
  
```
Do we set this set_se_hidden() for instant dropped cols as well? if so, where? :thinking:

Reply:
decided to follow a different approach by removing dropped columns from DD so the entry from DD is in sync with the index object.